packages/babel-plugin-transform-typescript/src/enum.ts

```diff
@@ -124,7 +125,29 @@ export function translateEnumValues(
           value = t.stringLiteral(constValue);
         }
       } else {
-        value = initializer;
+        const IdentifierVisitor: Visitor = {
```

Comment:
Please move this visitor to the top-level of the file. We do some processing+validation to visitors, and if it's always the same object this work can be cached. You can get the `seen` map by using the "state" parameter:
```js
          Identifier(expr, { seen }) {
            // ...
```

(style nit: we always call non-classes with a name that starts with a lowercase letter. Maybe `enumSelfReferenceVisitor`.